connectors/sources/github_graphql.py

```diff
@@ -0,0 +1,889 @@
+                ],
+                "order": 1,
+                "type": "str",
+                "value": GITHUB_SERVER,
+            },
+            "host": {
+                "depends_on": [{"field": "data_source", "value": GITHUB_SERVER}],
+                "label": "GitHub URL",
+                "order": 2,
+                "type": "str",
+                "value": "http://127.0.0.1:5000",
+            },
+            "token": {
+                "label": "GitHub Token",
+                "order": 3,
+                "sensitive": True,
+                "type": "str",
+                "value": "changeme",
+            },
+            "repositories": {
+                "display": "textarea",
+                "label": "List of repositories",
+                "order": 4,
+                "type": "list",
+                "value": WILDCARD,
+            },
+            "ssl_enabled": {
+                "display": "toggle",
+                "label": "Enable SSL",
+                "order": 5,
+                "type": "bool",
+                "value": False,
+            },
+            "ssl_ca": {
+                "depends_on": [{"field": "ssl_enabled", "value": True}],
+                "label": "SSL certificate",
+                "order": 6,
+                "type": "str",
+                "value": "",
+            },
+            "retry_count": {
+                "display_value": RETRIES,
+                "display": "numeric",
+                "label": "Maximum retries per request",
+                "order": 7,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": RETRIES,
+                "validations": [{"type": "less_than", "constraint": 10}],
+            },
+        }
+
+    async def get_invalid_repos(self):
+        try:
+            self._logger.debug(
+                "Checking if there are any inaccessible repositories configured"
+            )
+            foreign_repos, configured_repos = [], []
+            self.github_client.user = await self.github_client.get_logged_in_user()
+
+            for repo_name in self.github_client.repos:
+                if repo_name not in ["", None]:
+                    if "/" in repo_name:
+                        foreign_repos.append(repo_name)
+                    else:
+                        configured_repos.append(
+                            f"{self.github_client.user}/{repo_name}"
+                        )
+            async for repo in self.github_client.get_user_repos():
+                self.user_repos[repo["nameWithOwner"]] = repo
+            invalid_repos = list(set(configured_repos) - set(self.user_repos.keys()))
+
+            for repo_name in foreign_repos:
+                try:
+                    self.foreign_repos[
+                        repo_name
+                    ] = await self.github_client.get_foreign_repo(repo_name=repo_name)
+                except Exception:
```

Comment:
Maybe it's worth to add a debug log line like `f"Detected invalid repository: {repo_name}"`